core/src/main/scala/com/banno/vault/Vault.scala

```diff
@@ -86,6 +86,27 @@ object Vault {
     }
   }
 
+  /**
+   *  https://www.vaultproject.io/api/secret/kv/kv-v1#list-secrets uses GET alternative https://www.vaultproject.io/api-docs#api-operations vs LIST
+   */
+  def listSecrets[F[_]](client: Client[F], vaultUri: Uri)(token: String, secretPath: String)(implicit F: Sync[F]): F[VaultKeys] = {
+    val newSecretPath = if (secretPath.startsWith("/")) secretPath.substring(1) else secretPath
+    val request = Request[F](
+        method = Method.GET,
+        uri = vaultUri.withPath(s"/v1/$newSecretPath").withQueryParam("list", "true"),
+        headers = Headers.of(Header("X-Vault-Token", token))
+      )
+    for {
+      result <- 
+        F.adaptError(client.expect[VaultKeys](request)(jsonOf[F, VaultKeys])) {
+          case InvalidMessageBodyFailure(_, Some(cause: DecodingFailure)) =>
+            InvalidMessageBodyFailure("Could not decode vault list secrets response", cause.some)
+        }.handleErrorWith { e =>
+          F.raiseError(VaultRequestError(request = request, cause = e.some, extra = s"tokenLength=${token.length}".some))
+        }
+    } yield result
```

Comment:
No need of for-comprehension if there is a single item. 
```suggestion
      F.adaptError(client.expect[VaultKeys](request)(jsonOf[F, VaultKeys])) {
        case InvalidMessageBodyFailure(_, Some(cause: DecodingFailure)) =>
          InvalidMessageBodyFailure("Could not decode vault list secrets response", cause.some)
      }.handleErrorWith { e =>
        F.raiseError(VaultRequestError(request = request, cause = e.some, extra = s"tokenLength=${token.length}".some))
      }
```